lib/gui/pages/settings/controllers/settings.js

```diff
@@ -53,34 +54,45 @@ module.exports = function(WarningModalService, SettingsModel) {
   this.model = SettingsModel;
 
   /**
-   * @summary Enable a dangerous setting
+   * @summary Toggle setting
    * @function
    * @public
    *
-   * @param {String} name - setting name
-   * @param {Object} options - options
-   * @param {String} options.description - modal description
-   * @param {String} options.confirmationLabel - modal confirmation label
+   * @description
+   * If warningOptions is given, it should be an object having `description` and `confirmationLabel`;
+   * these will be used to present a user confirmation modal before enabling the setting.
+   *
+   * @param {String} setting - setting key
+   * @param {Object} warningOptions - (optional) WarningModalService options
    * @returns {Undefined}
    *
    * @example
-   * SettingsController.enableDangerousSetting('unsafeMode', {
+   * SettingsController.toggle('unsafeMode', {
    *   description: 'Don\'t do this!',
    *   confirmationLabel: 'Do it!'
    * });
    */
-  this.enableDangerousSetting = (name, options) => {
-    if (!this.currentData[name]) {
-      this.model.set(name, false);
-      return this.refreshSettings();
+  this.toggle = (setting, warningOptions) => {
+
+    const value = this.currentData[setting];
+    const dangerous = !_.isUndefined(warningOptions);
+
+    AnalyticsService.logEvent('Toggle setting', {
+      setting,
+      value,
+      dangerous
+    });
+
+    if (!value || !dangerous) {
+      return this.model.set(setting, value);
     }
 
     // Keep the checkbox unchecked until the user confirms
-    this.currentData[name] = false;
+    this.currentData[setting] = false;
 
-    WarningModalService.display(options).then((userAccepted) => {
+    WarningModalService.display(warningOptions).then((userAccepted) => {
```

Comment:
I just realised we should be adding a `.catch(ErrorService.reportException)` to this promise, and add the error module to the dependencies.